pygam/pygam.py

```diff
@@ -87,22 +87,74 @@
              'diffs': Diffs,
              'accuracy': Accuracy,
              'coef': Coef
-            }
+             }
 
 PENALTIES = {'auto': 'auto',
              'derivative': derivative,
              'l2': l2,
              'none': none,
-            }
+             }
 
 CONSTRAINTS = {'convex': convex,
                'concave': concave,
                'monotonic_inc': monotonic_inc,
                'monotonic_dec': monotonic_dec,
                'circular': circular,
                'none': none
-              }
-
+               }
+
+
+def _parallel_inc(gam, X, y, weights):
+    n = len(y)
+    R = []
+    f = []
+    r = 0.
+    vars_ = defaultdict(list)
+    pool = Pool()
+    block_results = []
+    for mask in gam._block_masks(n):
+        Xk = gam._modelmat(X[mask])
```

Comment:
@maorn i suspect that we will need to move the line:
`Xk = gam._modelmat(X[mask])`
into the `process_one` function. this way we avoid forming.holding the entire model matrix in memory at any given time.

Reply:
i am not sure how do you avoid it, since all the structures are in memory outside this code, there shouldn't be any structure(matrix)  coping so it won't explode the memory 
you can change the code here if needed

Reply:
@maorn ahh i think i was a little sleepy when i wrote this comment :)

every iteration in the loop is overwriting `Xk` so we are never keeping all the `Xk`s in memory at the same time. i guess it doesnt matter too much if this line goes inside or outside of the parallel loop... however, since `gam._modelmat()` has many non-linear algebra operations we indeed should see performance gains from parallelizing.